props/saber_BC_buttons.h

```diff
@@ -2042,6 +2064,154 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+      int noblade_level_before = current_config->ohm / NO_BLADE;
+      FindBladeAgain();
+      int noblade_level_after = current_config->ohm / NO_BLADE;
+
+      if (noblade_level_before < noblade_level_after) {
+        if (!SaberBase::IsOn()) {
+        SaberBase::DoBladeDetect(false);
+        } else {
+          SaberBase::DoNewFont();
+        }
+      } else if(noblade_level_before > noblade_level_after) {
+        if (!SaberBase::IsOn()) {
+          SaberBase::DoBladeDetect(true);
+        } else {
+          SaberBase::DoNewFont();
+        }
+      } else {
+        if (prev_real_best_config != last_real_best_config) {
+        // A real blade change has occurred, play bladein/out.
+        // If bladein/out doesn't exist, hybrid_font handles playing font.wav instead.
+          PVLOG_NORMAL << "---REAL BLADE CHANGE DETECTED--- Playing bladein/out\n";
+          hybrid_font.SB_BladeDetect(!(id() >= NO_BLADE));
+          prev_real_best_config = last_real_best_config;
+          last_real_best_config = real_best_config;
+        } else {
+          PVLOG_NORMAL << "**** No Blade Change Detected, something else changed Blade ID somehow. Playing font.wav\n";
+          SaberBase::DoNewFont();
+        }
+      }
+    }
+  }
+
+  size_t FindBestConfig(bool announce = false) override {
+    // Get the OS determined best_config
+    real_best_config = PropBase::FindBestConfig(announce);
+
+    // Handle manual blade array switching mode
+    if (manual_blade_array_active) {
+      if (real_best_config != last_real_best_config) {
+        PVLOG_NORMAL << "---REAL BLADE CHANGE DETECTED--- EXITING manual mode\n";
+        manual_blade_array_active = false;
+        prev_real_best_config = last_real_best_config;
+        last_real_best_config = real_best_config;
+
+        find_blade_again_pending_ = true;
+        PollScanId();
+        return real_best_config;
+      }
+      // Spoof best_config during manual mode
+      fake_best_config = current_config - blades;  // the incremented array we manually set in FakeFindBladeAgain()
+      return fake_best_config;
+    }
+
+    // Normal mode: Update stored real best_config
+    if (real_best_config != last_real_best_config) {
+      last_real_best_config = real_best_config; // Keep track of the latest real config
+    }
+    return real_best_config; // Return the real config
+  }
+  #endif  // BLADE_ID_SCAN_MILLIS
+
+  void TriggerBladeID() {
+    FindBladeAgain();
+    if (SFX_array) {
+      SFX_array.Select(current_config - blades);
+      hybrid_font.PlayPolyphonic(&SFX_array);
+    } else {
+      SaberBase::DoNewFont();
+    }
+  }
+
+  void NextBladeArray() {
+    manual_blade_array_active = true; // Enable manual mode
+    PVLOG_NORMAL << "** Manually Switching to Blade Array: " << (current_config - blades + 1) << "\n";
+    FakeFindBladeAgain();
+  }
+
+  // Manual Blade Array Selection version of FindBladeAgain()
+  void FakeFindBladeAgain() {
```

Comment:
Seems like it would be easier to just spoof the blade ID and then call the real FindBladeAgain()

Reply:
Pretty sure I tried simple.
Activation and deactivation of blades i think were the issue. Possibly something else but I needed to duplicate some.
Sorry, a little rusty on this one at first glance.
I'l re-read the thread when this was going on to refresh.

Reply:
> I'm going to see if I can come up with a simpler way.

That's great